ocs_ci/ocs/ocp.py

```diff
@@ -530,6 +537,87 @@ def check_name_is_specified(self, resource_name=''):
                 "Resource name has to be specified in class!"
             )
 
+    def check_function_supported(self, support_var):
+        """
+        Check if the resource supports the functionality based on the
+        support_var.
+
+        Args:
+            support_var (bool): True if functionality is supported, False
+                otherwise.
+
+        Raises:
+            NotSupportedFunctionError: If support_var == False
+
+        """
+        if not support_var:
+            raise ResourceNameNotSpecifiedException(
+                "Resource name doesn't support this functionality!"
+            )
+
+    def check_phase(self, phase):
+        """
+        Check phase of resource
+
+        Args:
+            phase (str): Phase of resource object
+
+        Returns:
+            bool: True if phase of object is the same as passed one, False
+                otherwise.
+
+        Raises:
+            NotSupportedFunctionError: If resource doesn't have phase!
+
+        """
+        self.check_function_supported(self._has_phase)
+        self.check_name_is_specified()
+        try:
+            data = self.get()
+        except CommandFailed:
+            log.info(f"Cannot find resource object {self.resource_name}")
+            return False
+        try:
+            current_phase = data['status']['phase']
+            log.info(
+                f"Resource {self.resource_name} is in phase: {current_phase}!"
+            )
+            return current_phase == phase
+        except KeyError:
+            log.info(
+                f"Problem while reading phase status of resource "
+                f"{self.resource_name}, data: {data}"
+            )
+        return False
+
+    @retry(ResourceInUnexpectedState, tries=4, delay=5, backoff=1)
+    def wait_for_phase(self, phase, timeout=300, sleep=5):
+        """
+        Wait till phase of resource is the same as required one passed in
+        the phase parameter.
+
+        Args:
+            phase (str): Desired phase of resource object
+            timeout (int): Timeout in seconds to wait for desired phase
+            sleep (int): Time in seconds to sleep between attempts
+
+        Raises:
+            ResourceInUnexpectedState: In case the resource is not in expected
+                phase.
+            NotSupportedFunctionError: If resource doesn't have phase!
```

Comment:
Same as above, the `ResourceNameNotSpecifiedException` raised from `check_name_is_specified()` should be also listed here, right?